pkg/controller/appsync/helpers.go

```diff
@@ -51,6 +56,20 @@ func (c *Controller) syncApps(ctx context.Context, apps *clients.AppsResponse) *
 			continue
 		}
 
+		if _, found := agencyUpdatedAlready[realm.RegionCode]; !found {
+			// Sync the realm level items.
+			realm.AgencyBackgroundColor = app.AgencyColor
```

Comment:
We should add a BeforeSave or sanity check to make sure this is `/\a#[0-9]{6}\Z/`

Reply:
added to realm BeforeSave